service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/bucket/BucketCloneRolesService.java

```diff
@@ -0,0 +1,95 @@
+package bio.terra.workspace.service.resource.controlled.flight.clone.bucket;
+
+import bio.terra.cloudres.google.storage.StorageCow;
+import bio.terra.stairway.FlightMap;
+import bio.terra.workspace.service.crl.CrlService;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ControlledResourceKeys;
+import com.google.api.client.util.Strings;
+import com.google.cloud.Identity;
+import com.google.cloud.Policy;
+import com.google.cloud.Role;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class BucketCloneRolesService {
```

Comment:
I feel like a static class manually taking in autowirable objects is an anti-pattern. It's like we're inverting the dependency inversion.

I'm looking for which package is the services package. We're in `bio.terra.workspace.service.resource.controlled.flight.clone.bucket` now.

Reply:
All `@Component` in `bio.terra.workspace.service` are either directly under `service` or under `service.resources`. So to me this feels misplaced as a "Service".  It doesn't seem like part of the service layer. Particularly since the scope is utility methods for a single flight.

So maybe my feeling of "misplaced" is more about how the name rather than that it is an `@Component`

Reply:
renamed to `BucketCloneRolesComponent`.